src/app/services/transaction/transaction-base/transaction-base.component.ts

```diff
@@ -13,29 +13,34 @@ export class TransactionBaseComponent {
     public deleteDisabled: boolean;
```

Comment:
`addDisabled` can and probably should be replaced with a check against `state` and the data by id. It's just for demo purposes anyway.

Reply:
`deleteDisabled` should be a check like so:
```typescript
/** Check item for unsaved deletion */
public isDeleted(id) {
    const state = this.transactions.getState(id);
    return state && state.type === TransactionType.DELETE;
}
This can be used for the class application (since it's actually useful logic in a real scenario) and for the edit button check,
```

Reply:
Same applies as `isEdited` check.